module/EntraBeta/Microsoft.Entra.Beta/Applications/Grant-EntraBetaMCPServerPermission.ps1

```diff
@@ -0,0 +1,259 @@
+
+        # Get service principals for the resolved clients
+        $clientSps = @()
+        foreach ($client in $clients) {
+            try {
+                $sp = Get-ServicePrincipal $client.AppId $client.Name
+                $clientSps += @{
+                    Sp = $sp
+                    Name = $client.Name
+                    IsCustom = $client.IsCustom
+                }
+                Write-Verbose "Found service principal for: $($client.Name)"
+            }
+            catch {
+                Write-Warning "Could not get service principal for $($client.Name) (App ID: $($client.AppId)): $($_.Exception.Message)"
+            }
+        }
+
+        if ($clientSps.Count -eq 0) {
+            throw "No MCP client service principals could be found or created."
+        }
+
+        Write-Host "Operating on $($clientSps.Count) MCP client(s): $($clientSps.Name -join ', ')" -ForegroundColor Cyan
+
+        # Determine target scopes
+        if ($Scopes -and $Scopes.Count -gt 0) {
+            # Validate specified scopes
+            $invalidScopes = $Scopes | Where-Object { $_ -notin $availableScopes }
+            if ($invalidScopes) {
+                throw "Invalid scopes (not available on resource): $($invalidScopes -join ', ')"
+            }
+            $targetScopes = $Scopes | Sort-Object -Unique
+            Write-Host "Granting specific scopes: $($targetScopes -join ', ')" -ForegroundColor Cyan
+        }
+        else {
+            # Grant all available scopes (default behavior)
+            $targetScopes = $availableScopes
+            Write-Host "Granting all available scopes: $($targetScopes -join ', ')" -ForegroundColor Cyan
+        }
+
+        # Apply the permission grants to all client service principals
+        $results = @()
+        foreach ($clientSp in $clientSps) {
+            try {
+                $grant = Set-ExactScopes -clientSpId $clientSp.Sp.Id -resourceSpId $resourceSp.Id -targetScopes $targetScopes
+                $results += @{
+                    Client = $clientSp.Name
+                    Grant = $grant
+                    Success = $true
+                    Error = $null
+                }
+            }
+            catch {
+                $results += @{
+                    Client = $clientSp.Name
+                    Grant = $null
+                    Success = $false
+                    Error = $_.Exception.Message
+                }
+            }
+        }
+
+        # Display results
+        $successCount = ($results | Where-Object Success).Count
+        $errorCount = ($results | Where-Object { -not $_.Success }).Count
+
+        Write-Host "`nResults Summary:" -ForegroundColor Yellow
+        Write-Host "Successfully processed: $successCount client(s)" -ForegroundColor Green
+        if ($errorCount -gt 0) {
+            Write-Host "Failed to process: $errorCount client(s)" -ForegroundColor Red
+        }
+
+        foreach ($result in $results) {
```

Comment:
is this a standard practice in Entra PS?
can the function just return the list of updated scopes?